### PR TITLE
pagination and swagger ui implemented v5.0

### PR DESCRIPTION
**Pull request by CodeWebWeaver**

### CODE REVIEW

File: src/main/java/com/parkhomovsky/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -21,38 +24,45 @@
...
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create new book", description = "Adding new book to database")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto book) {
         return bookService.create(book);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get book by id", description = "Try to find book with provided id")
```
Comment (JeniaSan) [inline src/main/java/com/parkhomovsky/bookstore/controller/BookController.java]: As for me, "Try to find book with provided id" sounds strange. "Find" logic on repository layer, on service and controller layer only "get". And "try" also looks not ok.

File: src/main/java/com/parkhomovsky/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -21,38 +24,45 @@
...
+    @Operation(summary = "Search books", description = "Search books by title or author")
     public List<BookDto> search(BookSearchParameters parameters) {
         return bookService.bookSearch(parameters);
     }
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.getAll();
+    @Operation(summary = "List book using pagination", description =
+            "List books on page using parameters")
```
Comment (JeniaSan) [inline src/main/java/com/parkhomovsky/bookstore/controller/BookController.java]: Parameters? I know what you mean, but I think it's better to remove it

File: src/main/java/com/parkhomovsky/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -21,38 +24,45 @@
...
+    public List<BookDto> findAll(Pageable pageable) {
+        return bookService.getAll(pageable);
     }
 
     @DeleteMapping("/{id}")
     @ResponseStatus(HttpStatus.NO_CONTENT)
+    @Operation(summary = "Delete book by id", description = "Delete book from database")
```
Comment (JeniaSan) [inline src/main/java/com/parkhomovsky/bookstore/controller/BookController.java]: Wrong description, you don't delete book from database (remember about soft delete)

**Review by JeniaSan — CHANGES_REQUESTED** [review]
Good solution, let's improve it!

**Review by JeniaSan — APPROVED** [review]
Good job!

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**